scripts/filter_expected_warnings.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+"""Script to filter out old warnings from doxygen."""
+
+import sys
+import re
+
+
+class DoxygenWarning(object):
+    """Doxygen warning class."""
+
+    def __init__(self, firstline, filename, warning):
+        self.firstline = firstline
+        self.filename = filename
+        self.warning = warning
+        self.otherlines = []
+
+    def equals_ignoring_path_and_line_number(self, other):
+        """Return true if warnings have same filename and warning message."""
+        if self.filename != other.filename:
+            return False
+        if self.warning != other.warning:
+            return False
+        return self.otherlines == other.otherlines
+
+    def print_with_prefix(self, prefix):
+        print(prefix + self.firstline)
+        for line in self.otherlines:
+            print(prefix + '  ' + line)
+
+
+class WarningsList(object):
+    """List of Doxygen warnings."""
+
+    def __init__(self, all_lines):
+        """Create a list of the warnings in this file."""
+        self.warnings_list = []
+        current = None
+
+        warning_start_expr = re.compile(r'[^:]+/([^:/]+):\d+: (warning.*$)')
+
+        for line in all_lines:
+            if line.isspace():
+                continue
+
+            # Allow comments in the list of expected warnings.
+            if line.startswith('#'):
+                continue
+
+            matched = warning_start_expr.match(line)
+            if matched:
+                filename = matched.group(1)
+                warning = matched.group(2)
+                current = DoxygenWarning(line.strip(), filename, warning)
+                self.warnings_list.append(current)
+            elif line.startswith('  '):
```

Comment:
:pick: It would make more sense to me if you would check if the line starts with '  ' first, and only if it doesn't would you try the regular expression, and if that doesn't match then you'd report an error. It would reduce the number of times you try the regular expression. But you said it runs very fast, so it's not important.